pkg/satellite/service.go

```diff
@@ -89,12 +103,27 @@ func (s *Service) GetUser(ctx context.Context, id uuid.UUID) (*User, error) {
 		return nil, err
 	}
 
-	user, err := s.store.Users().Get(ctx, id)
+	return s.store.Users().Get(ctx, id)
+}
+
+// GetCompany returns company by userID
+func (s *Service) GetCompany(ctx context.Context, userID uuid.UUID) (*Company, error) {
+	token, ok := auth.GetAPIKey(ctx)
+	if !ok {
+		return nil, errs.New("no api key was provided")
+	}
+
+	claims, err := s.authenticate(string(token))
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.authorize(ctx, claims)
```

Comment:
The above code (GetAPIKey->authenticate->authorize) is duplicated with other methods like `GetUser`. How about changing `s.authorize` to make all this work by taking just the `ctx` context?